### PR TITLE
GitHub Actions Improvements

### PR DESCRIPTION
**Pull request by rtwalker**
Adding a few commits of things I've thought might improve our GitHub Actions setup while I spent some time this week reading through the documentation. I think the changes are mostly superficial (except I caught one place we still have `master` instead of `main`), but thought I'd offer to contribute them nevertheless.

### CODE REVIEW

**Comment by rtwalker** [comment]
@epgts, this probably conflicts with some of the stuff you're doing, so I'm happy to wait until that's merged and try to resolve myself.

**Comment by jerryxwu** [comment]
@rtwalker Glad to see you taking the initiative to make it more maintainable. I noticed this when I was working on the other Github action.

We also actually made the switch from 'master' to 'main' branch in a release branch IIRC. The change did not somehow made to the main branch.

**Review by epgts — COMMENTED** [review]
Commit dbb1fdc gets my whole-hearted approval: I love it and thank you!

Commit 7d30768 gets half-hearted approval: I'm wondering what this setting buys us at all given that it was naming a non-existent branch with seemingly no harm done.

Thanks!

File: .github/workflows/clippy.yml
Comment (rtwalker) [reply .github/workflows/clippy.yml:32]: > Oof. I guess it was too much to hope that we could define all this in ONE place. Maybe we can file a feature request with github to implement templates? :)

I think we might be able to do this with [Composite Actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action). I'll try it out tomorrow.

Comment (rtwalker) [reply .github/workflows/clippy.yml:33]: de-duplicated the TODOs, leaving consolidating the `env:` declarations for another time.

**Review by epgts — APPROVED** [review]

**Comment by rtwalker** [comment]
bors retry

**Comment by rtwalker** [comment]
bors retry

File: .github/workflows/ci.yml
Comment (JLockerman) [reply .github/workflows/ci.yml:6]: @rtwalker Yup, you got it; this was set back when the branch was called `master` and before we hand branch protections. I _think_ what bors does isn't a push, so it shouldn't matter what's here.

File: .github/workflows/ci.yml
Comment (JLockerman) [inline .github/workflows/ci.yml:39]: I believe these are currently unused

**Review by JLockerman — APPROVED** [review]
I realized I never recorded my approval here: LGTM 👍

**Comment by rtwalker** [comment]
bors retry

**Pull request merged**
